### PR TITLE
ClangImporter: share more of the clang invocation

### PR DESCRIPTION
**Pull request by compnerd**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

This shares more of the clang invocation options across the various targets.  It
makes it easier to see what options actually differ.  Take the opportunity to
reorganise the options to be clustered by purpose.

### CODE REVIEW

**Comment by compnerd** [comment]
CC @jrose-apple

**Comment by jrose-apple** [comment]
I think the fact that Linux and Darwin both use `-std=gnu11` is a coincidence, not something that's inherently shared. The other options are things that the importer depends on in general. (In particular, one might get C++ support before the other.)

**Comment by compnerd** [comment]
It seems like it would be nicer to keep all the targets in as much parity as possible.  I was thinking of actually proposing to require libBlocksRuntime on Linux, and pushing `-fblocks` into the shared path as well.  It seems that it would be possible to share the entire invocation and just gate `-x objective-c -fobjc-arc` on ObjCInterop being enabled.

**Comment by compnerd** [comment]
@jrose-apple ping

**Comment by jrose-apple** [comment]
Sorry, I'm not convinced. Apple platforms really are different from everything else, and they will continue to be for a long time to come. I like the reorganization of the flags that are truly common, but the rest of them are all things we depend on for correctness in all configurations.

**Comment by compnerd** [comment]
@jrose-apple okay, well, this just reorders the arguments to be more logically grouped that you were still in favor of.

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -384,8 +387,6 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
     invocationArgStrs.insert(invocationArgStrs.end(), {
       // Non-Darwin platforms don't use the Objective-C runtime, so they can
       // not import Objective-C modules.
-      //
-      // Just use the most feature-rich C language mode.
```
Comment (jrose-apple) [inline lib/ClangImporter/ClangImporter.cpp]: This comment is still relevant, isn't it?

Comment (compnerd) [inline lib/ClangImporter/ClangImporter.cpp]: Ah, right, that should remain.  Ill update that shortly.

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Comment by compnerd** [comment]
@jrose-apple any other comments on this?

**Comment by jrose-apple** [comment]
Nope, all good!

**Pull request merged**